pkg/syslLogger/loggers/loggerInterface.go

```diff
@@ -0,0 +1,22 @@
+package loggers
+
+type Logger interface {
+	Debug(args ...interface{})
+	Debugf(format string, args ...interface{})
+	Error(args ...interface{})
+	Errorf(format string, args ...interface{})
+	Exit(code int)
+	Fatal(args ...interface{})
+	Fatalf(format string, args ...interface{})
+	Panic(args ...interface{})
+	Panicf(format string, args ...interface{})
+	Print(args ...interface{})
+	Printf(format string, args ...interface{})
+	Trace(args ...interface{})
+	Tracef(format string, args ...interface{})
+	Warn(args ...interface{})
+	Warnf(format string, args ...interface{})
+	WithField(key string, val interface{}) Logger
+	WithFields(fields map[string]interface{})
```

Comment:
The idea for `WithField` to return a `Logger` is so that user can do a chain to add the fields. Something like this.
```go
logger.WithField("val1", 1).WithField("val2", 2).WithField("val3", 3).Debug("This is a message")
```

So for `WithFields` to return a `Logger` seems pointless. Would it be better to return the `Logger` anyway?

Reply:
Will change it to return a new `Logger`

Reply:
changed to `PutField` and `PutFields`